tests/drivers/mqtt_transport.spec.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @boringnode/bus
+ *
+ * @license MIT
+ * @copyright Boring Node
+ */
+
+import { setTimeout } from 'node:timers/promises'
+import { test } from '@japa/runner'
+import { HiveMQContainer, StartedHiveMQContainer } from '@testcontainers/hivemq'
+import { MqttTransport } from '../../src/transports/mqtt.js'
+import { JsonEncoder } from '../../src/encoders/json_encoder.js'
+import { TransportEncoder, TransportMessage } from '../../src/types/main.js'
+
+test.group('Mqtt Transport', (group) => {
+  let container: StartedHiveMQContainer
+
+  group.setup(async () => {
+    container = await new HiveMQContainer().start()
+
+    return async () => {
+      await container.stop()
+    }
+  })
+
+  test('transport should not receive message emitted by itself', async ({ assert, cleanup }) => {
+    const transport = new MqttTransport({
+      host: container.getHost(),
+      port: container.getPort(),
+    }).setId('bus')
+    cleanup(() => transport.disconnect())
+
+    await transport.subscribe('testing-channel', () => {
+      assert.fail('Bus should not receive message emitted by itself')
+    })
+
+    await transport.publish('testing-channel', 'test')
+    await setTimeout(1000)
+  }).disableTimeout()
+
+  test('transport should receive message emitted by another bus', async ({
+    assert,
+    cleanup,
+  }, done) => {
+    assert.plan(1)
+
+    const transport1 = new MqttTransport({
+      host: container.getHost(),
+      port: container.getPort(),
+    }).setId('bus1')
+    const transport2 = new MqttTransport({
+      host: container.getHost(),
+      port: container.getPort(),
+    }).setId('bus2')
+
+    cleanup(async () => {
+      await transport1.disconnect()
+      await transport2.disconnect()
+    })
+
+    await transport1.subscribe('testing-channel', (payload) => {
+      assert.equal(payload, 'test')
+      done()
+    })
+
+    await setTimeout(200)
+
+    await transport2.publish('testing-channel', 'test')
+  }).waitForDone()
+
+  test('message should be encoded and decoded correctly when using JSON encoder', async ({
+    assert,
+    cleanup,
+  }) => {
+    const transport = new MqttTransport(
+      {
+        host: container.getHost(),
+        port: container.getPort(),
+      },
+      new JsonEncoder()
+    ).setId('bus')
+    cleanup(() => transport.disconnect())
+
+    const data = { test: 'test' }
+
+    await transport.subscribe('testing-channel', (payload) => {
+      assert.deepEqual(payload, data)
```

Comment:
this test is invalid : If the `subscribe` is never called then the assertion will never be executed and the test will create a false positive. We should use `done` or the assertion plannings

https://japa.dev/docs/plugins/assert#plan

Reply:
So, is the `"message should be encoded and decoded correctly when using JSON encoder"` test from Redis also invalid? The test doesn't check `assert.plan(1)` because there is no `done()` call, and it uses only one transport, which can't receive its proper data.

Reply:
This one ? 
https://github.com/boringnode/bus/blob/main/tests/drivers/redis_transport.spec.ts#L80

Yeah, definitely invalid too. This line will never be called https://github.com/boringnode/bus/blob/main/tests/drivers/redis_transport.spec.ts#L92 

and this is a good example of what I was trying to explain. The test passes even though the assertion was never called